feat: enable inline config editing & improve UI

GGalya1 opened the pull request:
Added predefined keys in the config for specific sensor types.
Added input fields for each configuration entry to allow editing.

Old version:
![image](https://github.com/user-attachments/assets/4e33276e-9ecd-47aa-9337-21c068d5bac2)
![image](https://github.com/user-attachments/assets/cf09f056-d5db-410f-a0bd-0f2243fd10d6)


New version:
![image](https://github.com/user-attachments/assets/89fa9c7d-9dec-4a10-b56b-f8f639ff81b3)
![image](https://github.com/user-attachments/assets/f2a15aa5-ffbd-4ab1-afdf-432972fffe7f)
![image](https://github.com/user-attachments/assets/83b7fb49-f843-4537-a8ec-874665071d2c)

GGalya1 committed:
feat: enable inline config editing & improve UI

- ConfigEditorModal now has a new field selectedSensorType, which passes the sensor type name to determine which fields to display.
- The configuration field now allows modifying values for each configuration key via an input field.
- Added new styles.

GGalya1 committed:
fix: changing the structure to fit Tinkerforge type sensors

GGalya1 committed:
added UID field to the table

GGalya1 committed:
hotfix: added uid field to the const "newPost""

GGalya1 committed:
Merge branch 'dev' into fix/missing-config-fields

GGalya1 committed:
fix: changed UID field type to number, and UID data from the database now displays correctly

anshulchahar reviewed (dismissed):
Please handle config and on_connect in another PR. 
UID is there.
We can merge it once CI pipeline starts working.

GGalya1 committed:
fix: config display fix

- PostList correctly calculates the height for each row based on the number of elements inside the config. If config-column is disabled the row is 170 heigh.
- in other cells of the table the data are carried over if they are not placed in one line (see TableItem.module.css)

GGalya1 committed:
fix: the UID column has been shifted to the end, as it contains not very important information

GGalya1 committed:
Merge branch 'dev' into fix/missing-config-fields

GGalya1 commented:
> Please handle config and on_connect in another PR. UID is there. We can merge it once CI pipeline starts working.

done

GGalya1 committed:
Merge branch 'dev' into fix/missing-config-fields

anshulchahar reviewed (approved):
looks fine
